File: sms-api-java/sendmessageservice/src/main/java/com/example/sendmessageservice/service/MessageService.java
package com.example.sendmessageservice.service;

import com.example.sendmessageservice.model.Message;

import java.util.List;

public interface MessageService {


    Message saveMessage (Message message);
    List<Message> getAllMessages();
    Message getMessageById(long id);
    void deleteMessage (long id);
    Message updateMessage(Message message);
}

File: sms-api-kotlin/sendmessageservice/src/main/kotlin/com/example/sendmessageservice/service/MessageService.kt
package com.example.sendmessageservice.service

import com.example.sendmessageservice.model.Message

interface MessageService {

    fun saveMessage(message: Message): Message
    fun getAllMessages(): List<Message>
    fun getMessageById(id: Long): Message
    fun deleteMessage(id: Long)
    fun updateMessage(id: Long, message: Message): Message
}